pkg/imager/extensions/rebuild.go

```diff
@@ -0,0 +1,85 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package extensions
+
+import (
+	"os"
+	"os/exec"
+)
+
+func (builder *Builder) rebuildInitramfs(tempDir string) error {
+	builder.Printf("creating system extensions initramfs archive and compressing it")
+
+	// the code below runs the equivalent of:
+	//   find $tempDir -print | cpio -H newc --create --reproducible | xz -v -C crc32 -0 -e -T 0 -z
+
+	listing, err := buildContents(tempDir)
+	if err != nil {
+		return err
+	}
+
+	pipeR, pipeW, err := os.Pipe()
+	if err != nil {
+		return err
+	}
+
+	defer pipeR.Close() //nolint:errcheck
+	defer pipeW.Close() //nolint:errcheck
+
+	// build cpio image which contains .sqsh images and extensions.yaml
+	cmd1 := exec.Command("cpio", "-H", "newc", "--create", "--reproducible")
```

Comment:
let's add a todo here to use the go equivalent, i think i used some when doing the extensions module support

Reply:
I'm not sure if it supports all the stuff we use here, but certainly interesting to make it fully in Go, this way we could ship it as a binary

Reply:
we also need the option for auto-enroll if it's uki

Reply:
in the initramfs?

Reply:
the auto-enroll files are in the EFI partition

Reply:
yes, I'm far away from it right now, this code installs system extensions in the initramfs :)

Reply:
just brought up something I remembered